server/src-lib/Hasura/GraphQL/Execute/Query.hs

```diff
@@ -196,51 +202,61 @@ convertQuerySelSet
      , Has UserInfo r
      , HasVersion
      , MonadIO m
+     , MonadIO tx
+     , MonadTx tx
      )
-  => HTTP.Manager
+  => Env.Environment
+  -> HTTP.Manager
   -> [N.Header]
   -> QueryReusability
   -> V.ObjectSelectionSet
   -> QueryActionExecuter
-  -> m (LazyRespTx, Maybe ReusableQueryPlan, GeneratedSqlMap)
-convertQuerySelSet manager reqHdrs initialReusability selSet actionRunner = do
+  -> m (tx EncJSON, Maybe ReusableQueryPlan, GeneratedSqlMap, [R.QueryRootFldUnresolved])
```

Comment:
Is this related to caching?

Reply:
This does appear to be related to caching. I think that maybe it was included accidentally when splitting the PR logically into project, env, and tracing/caching.